project14-Whack-a-Penguin/WhackSlot.swift

```diff
@@ -0,0 +1,79 @@
+//
+//  WhackSlot.swift
+//  project14-Whack-a-Penguin
+//
+//  Created by Felipe Gil on 2021-08-05.
+//
+
+import UIKit
+import SpriteKit
+
+class WhackSlot: SKNode {
+    var charNode: SKSpriteNode!
+    var isVisible = false
+    var isHit = false
+    
+    
+    func configure(at position: CGPoint) {
+        self.position = position
+        
+        let sprite = SKSpriteNode(imageNamed: "whackHole")
+        addChild(sprite)
+        
+        let cropNode = SKCropNode()
+        cropNode.position = CGPoint(x: 0, y: 15)
+        cropNode.zPosition = 1
+        cropNode.maskNode = SKSpriteNode(imageNamed: "whackMask")
+        
+        charNode = SKSpriteNode(imageNamed: "penguinGood")
+        charNode.position = CGPoint(x: 0, y: -90)
+        cropNode.addChild(charNode)
+        
+        addChild(cropNode)
+    }
+    
+    func show(hideTime: Double) {
+        if isVisible { return }
+        charNode.xScale = 1
+        charNode.yScale = 1
+        
+        charNode.run(SKAction.moveBy(x: 0, y: 80, duration: 0.05))
+        isVisible = true
+        isHit = false
+        
+        if Int.random(in: 0...2) == 0 {
+            charNode.texture = SKTexture(imageNamed: "penguinGood")
+            charNode.name = "charFriend"
+        } else {
+            charNode.texture = SKTexture(imageNamed: "penguinEvil")
+            charNode.name = "charEnemy"
+        }
+        
+        DispatchQueue.main.asyncAfter(deadline: .now() + (hideTime * 3.5)) {
+        [weak self] in
+            guard let self = self else { return }
+            self.hidePenguin()
```

Comment:
This could be just `self?.hidePenguin()` as well.

Reply:
Fixed =)